src/component/NavBar.js

```diff
@@ -1,27 +1,61 @@
-import React from "react";
-import { Nav, NavItems, NavLinks, NavTitle } from "../Styles/StyledComponents";
+import React, { Component } from "react";
+import {
+  Nav,
+  NavItems,
+  NavLinks,
+  NavTitle,
+  MenuIcon
+} from "../Styles/StyledComponents";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faAlignJustify } from "@fortawesome/free-solid-svg-icons";
+import NavMobileMenu from "./NavMobileMenu";
+import Backdrop from "./Backdrop";
 
-const NavBar = () => (
-  <Nav>
-    <NavTitle>CodePurpose</NavTitle>
-    <NavItems>
-      <li>
-        <NavLinks href="/">Home</NavLinks>
-      </li>
-      <li>
-        <NavLinks href="/">About</NavLinks>
-      </li>
-      <li>
-        <NavLinks href="/">Blog</NavLinks>
-      </li>
-      <li>
-        <NavLinks href="/">Careers</NavLinks>
-      </li>
-      <li>
-        <NavLinks href="/">Contact</NavLinks>
-      </li>
-    </NavItems>
-  </Nav>
-);
+class NavBar extends Component {
+  state = { mobileMenuOpen: false };
+
+  menuToggleHandler = () => {
+    this.setState(prevState => {
+      return { mobileMenuOpen: !prevState.mobileMenuOpen };
+    });
+  };
+
+  backdropHandler = () => {
+    this.setState({ mobileMenuOpen: false });
+  };
+
+  render() {
+    const { mobileMenuOpen } = this.state;
+    return (
+      <div>
+        <Nav>
+          <NavTitle>CodePurpose</NavTitle>
+          <NavItems>
+            <li>
+              <NavLinks href="/">Home</NavLinks>
+            </li>
+            <li>
+              <NavLinks href="/">About</NavLinks>
+            </li>
+            <li>
+              <NavLinks href="/">Blog</NavLinks>
+            </li>
+            <li>
+              <NavLinks href="/">Careers</NavLinks>
+            </li>
+            <li>
+              <NavLinks href="/">Contact</NavLinks>
+            </li>
+          </NavItems>
+          <MenuIcon onClick={this.menuToggleHandler}>
+            <FontAwesomeIcon icon={faAlignJustify} size="2x" />
+          </MenuIcon>
+        </Nav>
+        <NavMobileMenu open={mobileMenuOpen} />
```

Comment:
I wonder if this is how some of the libs implement their mobile hamburger menus? I had always just thought that they made their main menus reactive and the browser could properly render the other navs.